### PR TITLE
Fixes #2198 - Pass in targetOrigin from intern config for tests

### LINKED ISSUE #2198: Remove hardcoded instances of "http://localhost:5000/" from functional tests
They should instead pull that in from intern `siteRoot` config property.

### PR DESCRIPTION
**Pull request by miketaylr**
(Rather than hard-code)

r? @magsout

### CODE REVIEW

File: tests/functional/image-uploads-non-auth.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ const { assert } = intern.getPlugin("chai");
 const { registerSuite } = intern.getInterface("object");
 const FunctionalHelpers = require("./lib/helpers.js");
 const url = intern.config.functionalBaseUrl + "issues/new";
+const targetOrigin = intern.config.functionalBaseUrl.slice(
```
Comment (miketaylr) [inline tests/functional/image-uploads-non-auth.js:11]: this is super verbose!

Comment (magsout) [reply tests/functional/image-uploads-non-auth.js]: like verbose const ;)

File: tests/functional/image-uploads-non-auth.js [hunk 2]
```diff
@@ -7,7 +7,9 @@ const intern = require("intern").default;
 const { assert } = intern.getPlugin("chai");
 const { registerSuite } = intern.getInterface("object");
 const FunctionalHelpers = require("./lib/helpers.js");
-const url = intern.config.functionalBaseUrl + "issues/new";
+const baseUrl = intern.config.functionalBaseUrl;
+const url = baseUrl + "issues/new";
+const targetOrigin = baseUrl.slice(0, baseUrl.length - 1);
```
Comment (magsout) [inline tests/functional/image-uploads-non-auth.js:12]: hum, why? to remove the last 0 or maybe an `/` ?

```
"http://localhost:5000".slice(0, "http://localhost:5000".length -1)

-> "http://localhost:500"
```

Comment (magsout) [reply tests/functional/image-uploads-non-auth.js:12]: `"functionalBaseUrl": "http://localhost:5000",` : https://github.com/webcompat/webcompat.com/blob/master/intern.json#L5

File: tests/functional/image-uploads-non-auth.js (hunk 1 above)
Comment (miketaylr) [reply tests/functional/image-uploads-non-auth.js]: heh

File: tests/functional/image-uploads-non-auth.js (hunk 2 above)
Comment (miketaylr) [reply tests/functional/image-uploads-non-auth.js:12]: yeah, so this is kind of weird, but when you access the value intern adds a terminating `/` -> `http://localhost:5000/`, and postMessage won't match on that. 

(maybe i should file a bug with intern...)

**Pull request merged**